frontend/model/contracts/chatroom.js

```diff
@@ -75,13 +76,20 @@ function messageReceivePostEffect ({ contractID, messageHash, datetime, text, is
   const rootGetters = sbp('state/vuex/getters')
   const isDirectMessage = rootGetters.isDirectMessage(contractID)
   const isDMOrMention = isMentionedMe || isDirectMessage
-
-  if (!isAlreadyAdded && (isDMOrMention || messageType === MESSAGE_TYPES.INTERACTIVE)) {
+  const type = messageType === MESSAGE_TYPES.INTERACTIVE
+    ? UNREAD_MESSAGE_TYPE.INTERACTIVE
+    : messageType === MESSAGE_TYPES.POLL
+      ? UNREAD_MESSAGE_TYPE.POLL
+      : messageType === MESSAGE_TYPES.TEXT && isDMOrMention
+        ? UNREAD_MESSAGE_TYPE.MENTION
+        : null
```

Comment:
Is there a way to simplify this so that it's not as ugly?

For example, can you change the code so that this works?

```js
const type = messageType
```

Reply:
Does it look ugly to you? Hahaha. I will simplify this surely.

Reply:
```
const type = messageType
    ? {
        [MESSAGE_TYPES.TEXT]: isDMOrMention ? MESSAGE_TYPES.TEXT : undefined,
        [MESSAGE_TYPES.INTERACTIVE]: MESSAGE_TYPES.INTERACTIVE,
        [MESSAGE_TYPES.POLL]: MESSAGE_TYPES.POLL
      }[messageType]
    : undefined
```

Is this much better?

Reply:
That code will throw an exception when `NOTIFICATION: 'notification'` is passed in.

Is it possible to use the message type directly instead of creating another (very similar) type?

Reply:
> That code will throw an exception when NOTIFICATION: 'notification' is passed in.

The type will be `undefined` when `messageType` is `MESSAGE_TYPES.NOTIFICATION`.

> Is it possible to use the message type directly instead of creating another (very similar) type?

Yes we can. But please see the next line of `if` statement which decides if we should call `addChatRoomUnreadMessage`.

Reply:
> The type will be `undefined` when `messageType` is `MESSAGE_TYPES.NOTIFICATION`.

Oops, you're right!

> Yes we can. But please see the next line of `if` statement which decides if we should call `addChatRoomUnreadMessage`.

Ah, I see. Ok yeah this code should be good. I think.

Also, I don't think you need the `?:` to check `messageType`. You can just directly index the object. `{ a: 5 }[undefined]` still returns `undefined`

Reply:
> The type will be undefined when messageType is MESSAGE_TYPES.NOTIFICATION.

Oops, I have  found an interesting issue.
I have tested the following three in the browser console and got an error for the first one unexpectedly.
```
{ name: 'Alex', age: 31 }['name'] // Uncaught SyntaxError
({ name: 'Alex', age: 31 })['name'] // 'Alex'
({ name: 'Alex', age: 31 })['height'] // undefined
```

Reply:
Hmm, ok I guess make sure to wrap it in parenthesis 🤔